src/Tokenizer/Tokens.php

```diff
@@ -1431,21 +1444,16 @@ private function registerFoundToken($token): void
 
     /**
      * Unregister token as not found.
-     *
-     * @param array{int}|string|Token $token token prototype
      */
-    private function unregisterFoundToken($token): void
+    private function unregisterFoundToken(int $index): void
     {
+        $token = $this[$index];
```

Comment:
Should we check if the token exists under specified index, and return early if not? Otherwise `assert()` below can be invalid 🤔?

Reply:
The index must be always present as the unregister method is private and called only when the index exists.

in https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/8234/commits/c797a53589c539865926b396f45d3cddf827d5f2 I have added an `assert` to state this explicitly

Reply:
I have reverted the `assert` as it does not help phpstan.

there should be no condition as calling this method with non-existing index is wrong and if we would add a condition, it would be a dead code

Reply:
If calling this method with a non-existing index is wrong, then we should throw custom exception like we do in many other places. This way we're sure the index is correct after the check (SA-friendly), but also we don't let SPL exception slip outside of our API.